packages/devtools_app/lib/src/screens/vm_developer/object_inspector/vm_object_pool_display.dart

```diff
@@ -0,0 +1,138 @@
+// Copyright 2022 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:flutter/material.dart';
+import 'package:vm_service/vm_service.dart';
+
+import '../../../shared/common_widgets.dart';
+import '../../../shared/primitives/utils.dart';
+import '../../../shared/split.dart';
+import '../../../shared/table/table.dart';
+import '../../../shared/table/table_data.dart';
+import '../vm_developer_common_widgets.dart';
+import '../vm_service_private_extensions.dart';
+import 'object_inspector_view_controller.dart';
+import 'vm_object_model.dart';
+
+abstract class _ObjectPoolColumnData extends ColumnData<ObjectPoolEntry> {
+  _ObjectPoolColumnData(super.title, {required super.fixedWidthPx});
+  _ObjectPoolColumnData.wide(super.title) : super.wide();
+
+  @override
+  bool get supportsSorting => false;
+}
+
+class _AddressColumn extends _ObjectPoolColumnData {
+  _AddressColumn()
+      : super(
+          'Offset',
+          fixedWidthPx: 160,
+        );
+
+  @override
+  int getValue(ObjectPoolEntry dataObject) {
+    return dataObject.offset;
+  }
+
+  @override
+  String getDisplayValue(ObjectPoolEntry dataObject) {
+    final value = getValue(dataObject);
+    return '[PP + 0x${value.toRadixString(16).toUpperCase()}]';
+  }
+}
+
+class _DartObjectColumn extends _ObjectPoolColumnData
+    implements ColumnRenderer<ObjectPoolEntry> {
+  _DartObjectColumn({required this.controller}) : super.wide('Object');
+
+  final ObjectInspectorViewController controller;
+
+  @override
+  Object getValue(ObjectPoolEntry entry) => entry.value;
+
+  @override
+  Widget? build(
+    BuildContext context,
+    ObjectPoolEntry data, {
+    bool isRowSelected = false,
+    VoidCallback? onPressed,
+  }) {
+    if (data.value is int) return Text(data.value.toString());
+    return VmServiceObjectLink<Response>(
+      object: data.value as Response,
+      onTap: (e) {
```

Comment:
`onTap` is a required, non-nullable parameter at this point. Also, I have an issue filed that tracks this since we need to add support for viewing `Sentinel`s, which aren't instances of `ObjRef`.